go-controller/pkg/ovn/acl/acl.go

```diff
@@ -71,83 +32,38 @@ func RemoveACLFromNodeSwitches(switches []string, aclUUID string) error {
-	}
-	klog.Infof("ACL: %s, removed from the port group : %s", aclUUID, clusterPortGroupUUID)
-	return nil
-}
-
-// AddRejectACLToPortGroup adds a reject ACL to a PortGroup
-func AddRejectACLToPortGroup(clusterPortGroupUUID, aclName, sourceIP string, sourcePort int, proto v1.Protocol) (string, error) {
-	l3Prefix := "ip4"
-	if utilnet.IsIPv6String(sourceIP) {
-		l3Prefix = "ip6"
-	}
-
-	aclMatch := fmt.Sprintf("match=\"%s.dst==%s && %s && %s.dst==%d\"", l3Prefix, sourceIP,
-		strings.ToLower(string(proto)), strings.ToLower(string(proto)), sourcePort)
-	cmd := []string{"--id=@reject-acl", "create", "acl", "direction=" + types.DirectionFromLPort, "priority=" + types.DefaultDenyPriority, aclMatch, "action=reject",
-		fmt.Sprintf("name=%s", aclName), "--", "add", "port_group", clusterPortGroupUUID, "acls", "@reject-acl"}
-	aclUUID, stderr, err := util.RunOVNNbctl(cmd...)
-	if err != nil {
-		return "", errors.Wrapf(err, "Failed to add ACL: %s, %q, to cluster port group %s, stderr: %q", aclUUID, aclName, clusterPortGroupUUID, stderr)
-	}
-	return aclUUID, nil
-}
-
-// AddRejectACLToLogicalSwitch adds a reject ACL to a logical switch
-func AddRejectACLToLogicalSwitch(logicalSwitch, aclName, sourceIP string, sourcePort int, proto v1.Protocol) (string, error) {
-	l3Prefix := "ip4"
-	if utilnet.IsIPv6String(sourceIP) {
-		l3Prefix = "ip6"
-	}
-
-	aclMatch := fmt.Sprintf("match=\"%s.dst==%s && %s && %s.dst==%d\"", l3Prefix, sourceIP,
-		strings.ToLower(string(proto)), strings.ToLower(string(proto)), sourcePort)
-	cmd := []string{"--id=@reject-acl", "create", "acl", "direction=" + types.DirectionFromLPort, "priority=" + types.DefaultDenyPriority, aclMatch, "action=reject",
-		fmt.Sprintf("name=%s", aclName), "--", "add", "logical_switch", logicalSwitch, "acls", "@reject-acl"}
-
-	aclUUID, stderr, err := util.RunOVNNbctl(cmd...)
-	if err != nil {
-		return "", errors.Wrapf(err, "Failed to add ACL: %s, %q, to cluster switch %s, stderr: %q", aclUUID, aclName, logicalSwitch, stderr)
+func PurgeRejectRules(nbClient libovsdbclient.Client) error {
+	acls := []nbdb.ACL{}
+	err := nbClient.WhereCache(func(acl *nbdb.ACL) bool {
+		return acl.Action == nbdb.ACLActionReject
+	}).List(&acls)
+	if err != nil && err != libovsdbclient.ErrNotFound {
+		return errors.Wrapf(err, "Error while querying ACLs with reject action: %v", err)
 	}
-	return aclUUID, nil
-}
 
-func PurgeRejectRules(clusterPortGroupUUID string) error {
-	data, stderr, err := util.RunOVNNbctl("--columns=_uuid", "--format=csv", "--data=bare", "--no-headings", "find", "acl", "action=reject")
-	if err != nil {
-		return errors.Wrapf(err, "Error while querying ACLs with reject action: %s", stderr)
-	}
-	if strings.TrimSpace(data) == "" {
+	if len(acls) == 0 {
 		klog.Info("No reject ACLs to remove")
 		return nil
 	}
 
-	for _, uuid := range strings.Split(data, "\n") {
-		err = RemoveACLFromPortGroup(uuid, clusterPortGroupUUID)
+	aclPtrs := []*nbdb.ACL{}
```

Comment:
I am using pointers in general in libovsdbops api because it seems the correct thing to do as we are moving these structs around a lot so we avoid copying data. Also libovsdb api uses pointers to the structs everywhere except on the List method so this inconsistency is kind of brought in by that api itself. When logical switch is migrated to libovsdb this code will look different anyway so I did not want to bother a lot with it.

Reply:
Actually would it just make sense to have a `FindRejectACLs` in `libovsdbops/acl.go'  that returns what we expect.